core/src/main/java/com/amplifyframework/analytics/AnalyticsUserProfile.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+/**
+ * Represents user specific data such as name, email, plan, location etc.
+ */
+public final class AnalyticsUserProfile {
+    private String name;
+    private String email;
+    private String plan;
+    private Location location;
+    private Properties customProperties;
+
+    public AnalyticsUserProfile(final String name,
```

Comment:
Which fields are `@Nullable`, `@NonNull`?

Implement `hashCode()`, `toString()`, and `equals()` for any DTOs like this, that would be used in production.

Can this be called just a `UserProfile`? Do we need to prefix everything with `AnalyticsUserProfile`? Are there other common `UserProfile` classes that it would clash with, if we call it that?

Reply:
Added null notations.